ush/python/pygfs/task/marine_bmat.py

```diff
@@ -38,18 +57,114 @@ def __init__(self, config):
         local_dict = AttrDict(
             {
                 'PARMsoca': os.path.join(self.task_config.PARMgfs, 'gdas', 'soca'),
+                'CALC_SCALE_EXEC': _calc_scale_exec,
                 'MARINE_WINDOW_BEGIN': _window_begin,
-                'MARINE_WINDOW_END': _window_end,
                 'MARINE_WINDOW_MIDDLE': self.task_config.current_cycle,
+                'MARINE_WINDOW_END': _window_end,
+                'MARINE_WINDOW_LENGTH': f"PT{self.task_config['assim_freq']}H",
                 'ENSPERT_RELPATH': _enspert_relpath,
-                'CALC_SCALE_EXEC': _calc_scale_exec,
-                'APREFIX': f"{self.task_config.RUN}.t{self.task_config.cyc:02d}z."
+                'APREFIX': f"{self.task_config.RUN}.t{self.task_config.cyc:02d}z.",
+                'OPREFIX': f"{self.task_config.RUN}.t{self.task_config.cyc:02d}z."
             }
         )
 
         # Extend task_config with local_dict
         self.task_config = AttrDict(**self.task_config, **local_dict)
 
+        # Create dictionary of Jedi objects
+        self.jedi = AttrDict()
+
+        # gridgen
+        self.jedi['gridgen'] = Jedi(AttrDict(
```

Comment:
Same here, though since I think these are called in the same job, you'll need to pass the correct configuration in when the each object is created.